tests/osn-tests/src/test_osn_global.ts

```diff
@@ -50,10 +57,92 @@ describe('osn_global', () => {
             try {
                 source = osn.Global.getOutputSource(1);
             } catch(e) {
-                throw new Error("failed to set source to output channel");
+                throw new Error(getCppErrorMsg(e));
             }
             
             expect(source.name).to.equal('test_osn_global_source');
         });
+
+        it('should fail if trying to get from empty output channel', () => {
+            let source: ISource;
+
+            try {
+                source = osn.Global.getOutputSource(5);
+            } catch(e) {
+                throw new Error(getCppErrorMsg(e));
```

Comment:
We have to ask our self if the way it fails is expected by the frontend. Throwing an error when something wrong happened is good but if nobody is handling that error then we just end up in the same bad state.

Reply:
The purpose of these tests is to guarantee that our functions are working correctly in the most simple test case possible. Here I'm throwing the error that comes from the backend to make the test case fail in case something went wrong on our side. I'm not factoring how the frontend is expected to behave in case of success or fails because it would increase the scope of the unit testing beyond what is being proposed by these tests as well as the time to have full test coverage.

Reply:
My point is that should we except that behavior in case of error? We should have a proper way to handle errors in the frontend and our backend should follow it. Then in our tests, we should test that our backend behaves correctly when an error occurred.

Reply:
Reworked all tests to be more in line with how the functions are used in the frontend. I'll have this in mind for all future tests